### ISSUE TITLE
Add "scale_factor" to Stereographic projections

### ISSUE THREAD

**Issue by perrette** [SciTools/cartopy#455]
Somehow related to Issue #339 

Is there a reason why no "scale_factor" is provided in Stereographic, NorthPolarStereo and SouthPolarStereo projections?

It does not appear explicitly in PROJ.4 call signature for [stereographic  projection](http://www.remotesensing.org/geotiff/proj_list/stereographic.html) but does in the table (as "Scale factor at natural origin") and is stated in the description "The GeoTIFF formula doesn't include a scale, but I don't see why it shouldn't. [...] so I am adding it to the official formulation [...]".

The [netCDF CF-1.4 convention](http://cfconventions.org/1.4.html#grid-mappings-and-projections) for Stereographic and Polar Stereographic also mentions it under  "scale_factor_at_projection_origin". For the polar stereographic projection, this is directly related to the "true_scale_latitude" (or "standard_parallel" in CF-1.4), but the scale factor expresses instead, and more generally, the distortion at the projection origin (lat=+/-90). In both cases this corresponds to a projection plane which is not tangent but secant to the Earth surface (while still being parallel to the equator). The true scale latitude represents the latitude of intersection (a circle). If you picture two points near (and symmetrical w.r.t) the North Pole, very close to each other, they can be considered on the tangent plane. The scale factor then becomes the ratio of distances between tangent and projection (secant) planes. Thales is enough to express it as a function of latitude (for a spherical Earth):
   
    scale_factor = (sin(true_scale_latitude) + 1) / 2
    
(also provided in the [netCDF-Java reference doc on coordinate transforms] (https://www.unidata.ucar.edu/software/thredds/current/netcdf-java/reference/StandardCoordinateTransforms.html))

Anyway, this parameter is called `+k_0` in PROJ.4 (look for "ko" or "k0" in the doc for the more complex, ellipsoidal formula in polar or oblique stereo). To re-use the examples provided in Issue #339:

    $ echo "10 10" | proj +proj=stere +lon_0=0 +lat_0=0
    2215070.44	2234184.70

    $ echo "10 10" | proj +proj=stere +lon_0=0 +lat_0=0 +k_0=0.9
    1993563.39	2010766.23

Near-equivalence between `+k_0` and `+lat_ts` in polar stereo (using `bc` to implement the formula above):

    $ echo "10 10" | proj +proj=stere +lon_0=0 +lat_0=90  +ellps=sphere  # null-case
    1856612.01	-10529369.96

    $ echo "10 10" | proj +proj=stere +lon_0=0 +lat_0=90 +lat_ts=45 +ellps=sphere # true scale latitude
    1584717.48	-8987379.43

    $ echo "10 10" | proj +proj=stere +lon_0=0 +lat_0=90 +k_0=`echo "(s(45*3.141592653/180)+1)/2" | bc -l`  +ellps=sphere # equivalent scale factor
    1584717.48	-8987379.43

Things differ a bit for an ellipsoidal Earth.

So in summary for this issue and #337, and to be in accordance with CF-1.4 conventions I think a `scale_factor` parameter should be present in the `Stereographic` class.

A simple workaround to keep the current implementation of `NorthPolarStereo` and `SouthPolarStereo` as subclasses of `Stereographic`, I suggest that `true_scale_latitude` should not be removed from the `Stereographic` class, but instead a warning could be issued when this parameter is provided with a latitude different from +90 or -90.

**Comment by gauteh** [comment]
Does the false_easting and false_northing correspond to x0 and y0 in the docs?

**Comment by gauteh** [comment]
:+1: This is what is preventing me from using the IBCAO with cartopy.

**Comment by gauteh** [comment]
(also defining the x and y limits of the projection)

**Comment by pelson** [comment]
Let's see if we can get a workaround projection definition in #618.

**Comment by gauteh** [comment]
Hm, I must have too fast ( again ). I think I can achieve a correct setup with `true_scale_latitude` and `set_xlim`/`set_ylim`. I've got a bunch of tests set up for my case, and I'll add a link to the tests here once they are commited somewhere online.

**Comment by pelson** [comment]
Definitely not averse to adding keywords if they have meaning and are not duplicating existing functionality. If you genuinely were able to solve the problem with the existing keywords, would you be interested in extending the cartopy documentation?
